src/Common/src/System/Security/Cryptography/DerSequenceReader.cs

```diff
@@ -107,6 +110,61 @@ internal void SkipValue()
             _position += contentLength;
         }
 
+        internal void ValidateAndSkipDerValue()
+        {
+            byte tag = PeekTag();
+
+            // If the tag is in the UNIVERSAL class
+            if ((tag & TagClassMask) == 0)
+            {
+                // Tag 0 is special ("reserved for use by the encoding rules"), but mainly is used
+                // as the End-of-Contents marker for the indefinite length encodings, which DER prohibits.
+                //
+                // Tag 15 is reserved.
+                //
+                // So either of these are invalid.
+
+                if (tag == 0 || tag == 15)
+                    throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
```

Comment:
Maybe? :smile: Most of the reader just throws CryptographicException with a message only, because it's emulating Windows CryptDecodeObject.  It's unclear if the data would be interesting from an attack perspective; but (Dictionary.Add's duplicate item exception notwithstanding) it's usually easier to add more information later than to take it away, so I want to stick with the "pretend we're native and don't have a way of really representing this" for now.